### PR TITLE
Add UrlFilter module

### PR DESCRIPTION
**Pull request by changlan**

### CODE REVIEW

File: core/modules/url_filter.cc [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    Flow flow;
+    flow.src_ip = ip->src;
+    flow.dst_ip = ip->dst;
+    flow.src_port = tcp->src_port;
+    flow.src_port = tcp->dst_port;
+
+    // Check if the flow is already blocked
+    std::unordered_map<Flow, uint64_t, FlowHash>::iterator it;
+    if ((it = blocked_flows_.find(flow)) != blocked_flows_.end()) {
+      uint64_t now = tsc_to_ns(rdtsc());
```
Comment (sangjinhan) [inline core/modules/url_filter.cc]: `rdtsc()` is an expensive operation. You can use `ctx.current_ns()`, which is refreshed for every scheduling round, in place of `tsc_to_ns(rdtsc())`

File: core/modules/url_filter.cc [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+#include "../utils/ether.h"
+#include "../utils/http_parser.h"
+#include "../utils/ip.h"
+
+using bess::utils::EthHeader;
+using bess::utils::Ipv4Header;
+using bess::utils::TcpHeader;
+
+const uint64_t TIME_OUT = 12e10;
```
Comment (sangjinhan) [inline core/modules/url_filter.cc]: `TIME_OUT_NS` might be better?
Also, if you intended 2 minutes, 120e9 would be a better representation

**Comment by changlan** [comment]
Sounds good. Will change.

File: core/modules/url_filter.cc (hunk 2 above)
Comment (sangjinhan) [inline core/modules/url_filter.cc]: ... or 120 * 1000 * 1000 * 1000. 120e9 is a floating point number so it may be slightly off after type casting.

Comment (changlan) [inline core/modules/url_filter.cc]: Got the "integer overflow in expression" error. Will use 10000000000 instead..

Comment (changlan) [inline core/modules/url_filter.cc]: Oh I could do `120L * 1000 * 1000 * 1000`. Nevermind.

**Comment by sangjinhan** [comment]
Merged, thanks!
i15573941050

**Pull request merged**